src/services/goals.js

```diff
@@ -83,9 +83,18 @@ async function removeUnusedObjectivesGoalsFromReport(reportId, currentGoals, tra
   const previousObjectiveIds = previousObjectives.map((ro) => ro.objectiveId);
 
   const goalIdsToRemove = previousGoalIds.filter((id) => !currentGoalIds.includes(id));
-  const objectiveIdsToRemove = previousObjectiveIds
-    .filter((id) => !currentObjectiveIds.includes(id));
-
+  const objectiveIdsToRemove = [];
+  Promise.all(
```

Comment:
Do we need to await this `promise.all`?

Reply:
It hasn't been a problem, but I will add it just to be safe.